src/partners/switchain.ts

```diff
@@ -0,0 +1,115 @@
+import { asArray, asObject, asString, asUnknown } from 'cleaners'
+import fetch from 'node-fetch'
+
+import { PartnerPlugin, PluginParams, PluginResult, StandardTx } from '../types'
+
+const asSwitchainTx = asObject({
+  status: asString,
+  appId: asString,
+  createdAt: asString,
+  pair: asString,
+  depositTxId: asString,
+  depositAddress: asString,
+  amountFrom: asString,
+  withdrawAddress: asString,
+  rate: asString
+})
+
+const asSwitchainResult = asObject({
+  orders: asArray(asUnknown)
+})
+
+const PAGE_LIMIT = 100
+const QUERY_LOOKBACK = 1000 * 60 * 60 * 24 * 4 // 4 days ago
+
+export async function querySwitchain(
+  pluginParams: PluginParams
+): Promise<PluginResult> {
+  const ssFormatTxs: StandardTx[] = []
+  let apiKey
+  let latestTimestamp = 0
+  if (typeof pluginParams.settings.latestTimestamp === 'number') {
+    latestTimestamp = pluginParams.settings.latestTimestamp
+  }
+
+  if (typeof pluginParams.apiKeys.apiKey === 'string') {
+    apiKey = pluginParams.apiKeys.apiKey
+  } else {
+    return {
+      settings: { latestTimestamp: latestTimestamp },
+      transactions: []
+    }
+  }
+
+  latestTimestamp -= QUERY_LOOKBACK
```

Comment:
This may go negative. It is better to apply the query look back to the check on line 86

Reply:
whats wrong with it going negative?  the only time latesttimestamp is being used is for a comparison between the current transactions timestamp, its not actually getting set anywhere.  The out gets set to the most recent transaction's timestamp so its unrelated.  I'll change it the query lookback to that line though it does take up less space in the code